### PR TITLE
Explicitly set kind version when running tests in CI

### PR DESCRIPTION
**Pull request by swiatekm-sumo**
Up until now we depended on the `kind` version shipped by Github, which opened us up to unexpected breakage whenever they upgraded. This has happened recently with an upgrade from `0.19.0` to `0.20.0`.

This change:
- explicitly sets the kind version in CI
- updates Node images for kind 0.20.0
- uses kind 0.17.0 to run tests for K8s <1.21, which kind 0.20.0 doesn't support out of the box
- updates kuttl to the latest released version

### CODE REVIEW

File: kind-1.19.yaml
Comment (pavolloffay) [inline kind-1.19.yaml:5]: If we are dropping testing for 1.19 and 1.20 then we can remove this file.

File: kind-1.20.yaml
Comment (pavolloffay) [inline kind-1.20.yaml:5]: If we are dropping testing for 1.19 and 1.20 then we can remove this file.

File: kind-1.19.yaml
Comment (swiatekm-sumo) [reply kind-1.19.yaml:5]: Instead of dropping them, we now use an older kind version.

**Review by TylerHelmuth — APPROVED** [review]

File: .github/workflows/e2e.yaml
Comment (jaronoff97) [inline .github/workflows/e2e.yaml:16]: Instead of adding these to the environment like this, we could use the matrix.include functionality of github actions. https://docs.github.com/en/actions/using-jobs/using-a-matrix-for-your-jobs#expanding-or-adding-matrix-configurations

This would let us specify exactly KIND version each kube version should be using.

**Review by jaronoff97 — COMMENTED** [review]
Just one thought on how to make this a bit cleaner without env vars.

**Review by pavolloffay — APPROVED** [review]
Nice  fix for the CI 🥳 

LGTM, but @jaronoff97 has a good comment :)

File: .github/workflows/e2e.yaml
Comment (swiatekm-sumo) [reply .github/workflows/e2e.yaml:16]: I considered doing that, but given that:
1. There would only ever be two kind versions
2. There's less duplication this way, you only change the kind version in one place if you want to upgrade

I decided to do it this way. I don't have a strong opinion on it though, the matrix approach is fine.

**Comment by pavolloffay** [comment]
@swiatekm-sumo should be a similar fix applied to a local setup of kind?

**Comment by swiatekm-sumo** [comment]
> @swiatekm-sumo should be a similar fix applied to a local setup of kind?

I think so, but I didn't want to hold this PR up with even more changes. We should treat kind (and kuttl as well) the same way as we treat other local tools - have a make target that installs them to `LOCALBIN`, and use that binary throughout the Makefile.

**Comment by TylerHelmuth** [comment]
I view this PR as a temporary solution until we drop 1.19 (and other old versions) entirely.  I'm in favor of getting this merged to unblock other work and then quickly bumping our min k8s version.

**Comment by pavolloffay** [comment]
I am merging this to fix the CI, additional fixes can be submitted in follow up PRs

**Comment by pavolloffay** [comment]
> I view this PR as a temporary solution until we drop 1.19 (and other old versions) entirely

@TylerHelmuth why do you see this as a temporary solution?

**Comment by TylerHelmuth** [comment]
@pavolloffay I see the permanent solution to be bumping our minimum supported k8s version which I expect to correlate with being able to use a single kind version.

**Pull request merged**
